backend/controllers/prisonerProfile/prisonerCellHistory.js

```diff
@@ -0,0 +1,87 @@
+const { serviceUnavailableMessage } = require('../../common-messages')
+const {
+  formatName,
+  stripAgencyPrefix,
+  formatTimestampToDateTime,
+  possessive,
+  sortByDateTime,
+  putLastNameFirst,
+} = require('../../utils')
+const {
+  app: { notmEndpointUrl: dpsUrl },
+} = require('../../config')
+
+module.exports = ({ elite2Api, logError, page = 0 }) => async (req, res) => {
+  const { offenderNo } = req.params
+
+  const getAgencyDetails = async cells => {
+    return Promise.all(
+      cells
+        .map(cell => cell.agencyId)
+        .filter((v, i, a) => a.indexOf(v) === i)
+        .map(agencyId => elite2Api.getAgencyDetails(res.locals, agencyId))
+    )
+  }
+
+  const extractLocation = (location, agencyId) => {
+    const withoutAgency = stripAgencyPrefix(location, agencyId)
+    if (withoutAgency.includes('RECP')) return 'Reception'
+    if (withoutAgency.includes('CSWAP')) return 'Cell swap'
+    return withoutAgency
+  }
+
+  try {
+    const { bookingId, firstName, lastName } = await elite2Api.getDetails(res.locals, offenderNo)
+    const { content: cells } = await elite2Api.getOffenderCellHistory(res.locals, bookingId, {
+      page,
+      size: 10000,
```

Comment:
This is temporary to return all data for now. Pagination will be properly implemented at a later stage.